main.tf

```
@@ -79,6 +79,7 @@ resource "aws_ec2_transit_gateway_vpc_attachment" "this" {
     var.tags,
     { Name = var.name },
     var.tgw_vpc_attachment_tags,
+    each.value.tags,
```

```suggestion
    try(each.value.tags, {}),
```
Since `tags` is optional.